src/abstract_charm.py

```diff
@@ -214,6 +224,25 @@ def wait_until_mysql_router_ready(self) -> None:
         else:
             logger.debug("MySQL Router is ready")
 
+    @abc.abstractmethod
+    def expose(self) -> None:
+        """Expose MySQL Router."""
+        pass
+
+    @abc.abstractmethod
+    def unexpose(self) -> None:
+        """Unexpose MySQL Router."""
+        pass
+
+    def reconcile_node_port(self) -> None:
```

Comment:
should this be in the abstract_charm.py file? it seems something that is specific to k8s, and if so, should be moved to kubernetes_charm.py